src/visualizations/config/adapters/dhis_highcharts/series/index.js

```diff
@@ -168,10 +168,21 @@ function getDefault(series, layout, isStacked, extraOptions) {
             seriesObj.groupPadding = 0
         }
 
+        const legendSet = extraOptions?.legendSets?.find(
+            legendSet =>
+                legendSet.id === metaData.items[seriesObj.id]?.legendSet
+        )
+
         // color
-        seriesObj.color = isYearOverYear(layout.type)
-            ? indexColorPatternMap[index]
-            : idColorMap[seriesObj.id]
+        if (isYearOverYear(layout.type)) {
```

Comment:
I guess at this point it is hard to add unit tests for the new code (or extract this to a separate function). Given that isn't feasible, this could be a case where a descriptive comment would help explain the logic.